scripts/install/RegisterTaskQueueServices.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017-2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoAdvancedSearch\scripts\install;
+
+use oat\oatbox\extension\InstallAction;
+use oat\oatbox\reporting\Report;
+use oat\tao\model\search\tasks\AddSearchIndexFromArray;
+use oat\tao\model\search\tasks\DeleteIndexProperty;
+use oat\tao\model\search\tasks\RenameIndexProperties;
+use oat\tao\model\search\tasks\UpdateClassInIndex;
+use oat\tao\model\search\tasks\UpdateDataAccessControlInIndex;
+use oat\tao\model\search\tasks\UpdateResourceInIndex;
+use oat\tao\model\taskQueue\Queue;
+use oat\tao\model\taskQueue\QueueDispatcher;
+use oat\tao\model\taskQueue\QueueDispatcherInterface;
+use oat\taoTaskQueue\scripts\tools\BrokerFactory;
+
+class RegisterTaskQueueServices extends InstallAction
```

Comment:
When we have a new event / task to advanced search, can we reuse this script or we will need to create a new one?

And if there is already a queue created with this name, what will happen?

Maybe we should separate it to another class and make sure we can control those cases, cause for sure it will happen.

Reply:
Refactoring done